File: app/src/main/java/com/akexorcist/googledirection/sample/MainActivity.java
package com.akexorcist.googledirection.sample;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;


public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_simple).setOnClickListener(this);
        findViewById(R.id.btn_transit).setOnClickListener(this);
        findViewById(R.id.btn_alternative).setOnClickListener(this);
        findViewById(R.id.btn_waypoints).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (id == R.id.btn_simple) {
            goToSimpleDirection();
        } else if (id == R.id.btn_transit) {
            goToTransitDirection();
        } else if (id == R.id.btn_alternative) {
            goToAlternativeDirection();
        } else if (id == R.id.btn_waypoints) {
            goToWaypointsDirection();
        }
    }

    public void goToSimpleDirection() {
        openActivity(SimpleDirectionActivity.class);
    }

    public void goToTransitDirection() {
        openActivity(TransitDirectionActivity.class);
    }

    public void goToAlternativeDirection() {
        openActivity(AlternativeDirectionActivity.class);
    }

    public void goToWaypointsDirection() {
        openActivity(WaypointsDirectionActivity.class);
    }

    public void openActivity(Class<?> cs) {
        startActivity(new Intent(this, cs));
    }
}

File: app/src/main/java/com/akexorcist/googledirection/sample/MainActivity.kt
package com.akexorcist.googledirection.sample

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.akexorcist.googledirection.sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val binding: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.buttonSimple.setOnClickListener { goToSimpleDirection() }
        binding.buttonTransit.setOnClickListener { goToTransitDirection() }
        binding.buttonAlternative.setOnClickListener { goToAlternativeDirection() }
        binding.buttonWaypoints.setOnClickListener { goToWaypointsDirection() }
    }

    private fun goToSimpleDirection() {
        openActivity(SimpleDirectionActivity::class.java)
    }

    private fun goToTransitDirection() {
        openActivity(TransitDirectionActivity::class.java)
    }

    private fun goToAlternativeDirection() {
        openActivity(AlternativeDirectionActivity::class.java)
    }

    private fun goToWaypointsDirection() {
        openActivity(WaypointsDirectionActivity::class.java)
    }

    private fun openActivity(cs: Class<*>) {
        startActivity(Intent(this, cs))
    }
}
